src/axom/core/memory_management.hpp

```diff
@@ -220,7 +226,14 @@ inline T* reallocate(T* pointer, std::size_t n) noexcept
 
   // Umpire 2.1.0 and above handles reallocate(0) natively
```

Comment:
Should we remove logic relating to versions of Umpire earlier than 2.1.2 ? (They're currently on version 5.0.1)

Perhaps we should create an issue to remind us to discuss this and decide if we still need this.

Reply:
Logged as #618